### ISSUE TITLE
Adding Multiple Data Iterators as Input in R

### ISSUE THREAD

**Issue by wwkong** [dmlc/mxnet#1269]
Similar to a previously raised [issue](https://github.com/dmlc/mxnet/issues/617), I'm having some trouble understanding how I can create a single data iterator for multiple datasets, each with different dimensions. To illustrate, here is a sample situation:

``` R
# Load packages and data
require(mxnet)
trainDataIter1 <- mx.io.CSVIter(data.csv="trainData1.csv", data.shape = 3,
                                label.csv="trainLabel.csv", label.shape = 10,
                                batch.size=10)
trainDataIter2 <- mx.io.CSVIter(data.csv="trainData2.csv", data.shape = c(3,3),
                                label.csv="trainLabel.csv", label.shape = 10,
                                batch.size=10)

# Create the symbolic neural network
in1 <- mx.symbol.Variable('data')
fc1 <- mx.symbol.FullyConnected(data=in1, num.hidden=10, name='fc1')
in2 <- mx.symbol.Variable('data2')
fc2 <- mx.symbol.FullyConnected(data=in2, num.hidden=10, name='fc2')
compose.net <- mx.apply(fc1, data=fc2, name="compose")
out <- mx.symbol.LogisticRegressionOutput(data=compose.net, name='softmax')
```

In the code above, I have created two CSV data iterators which share a common labeling dataset but have different training data. In the symbolic network, let's say that the first iterator is used in fc1 and the second in fc2. I have also composed the two datasets before loading them into a softmax node.

Unfortunately, when I try to find a function which can accept multiple iterators in R, I am stuck. It seems that I must have an iterator which iterates over other iterators. I know in Python this is accomplished through the PrefetchingIter class and the function mx.io.PrefetchingIter function but no such function exists in R.

Is there a quick way in R to accomplish the same thing in Python?

**Comment by thirdwing** [comment]
Sorry for this, I should have time to add `PrefetchingIter` this weekend.

I have a deadline on Friday. So please be patient. I will really appreciate that.

**Comment by wwkong** [comment]
Any updates on this feature request? Is there anything I can do to help?

**Comment by jsusskin** [comment]
I have a similar question for the case where I'm using MXDataIter objects. I want to load in two different iterators and then route them to different symbolic input nodes. It looks like the NDArrayIter can help here but requires the inputs to be NDArrays, and I don't see how to combine that with MXDataIter objects. Can the PrefetchingIter implementation help here, or maybe this requires writing a custom iterator?

**Comment by EasonD3** [comment]
Any follow-ups? Would love to see how to do multiple inputs.

**Comment by qiaohaijun** [comment]
I occur the same problem

**Comment by MichelV2** [comment]
Any updates on this? This feature would be really nice in R

**Comment by thirdwing** [comment]
I think this can be solved by #6673 , where you can define your own iterator and put into the training process.
